wasm/jit/jit_amd64.go

```diff
@@ -1886,8 +1912,593 @@ func (c *amd64Compiler) compileI32WrapFromI64() error {
+	checkIfExceedsLowerBound.To.Reg = source.register
+	c.addInstruction(checkIfExceedsLowerBound)
+
+	// Jump if the value is -Inf.
+	jmpIfExceedsLowerBound := c.newProg()
+	jmpIfExceedsLowerBound.As = x86.AJCS
+	jmpIfExceedsLowerBound.To.Type = obj.TYPE_BRANCH
+	c.addInstruction(jmpIfExceedsLowerBound)
+
+	// At this point, the value is the minimum signed 32-bit int (=-2147483648.000000) or larger than 32-bit maximum.
+	// So, check if the value equals the minimum signed 32-bit int.
+	checkIfMinimumSignedInt := c.newProg()
+	if isFloat32Bit {
+		checkIfMinimumSignedInt.As = x86.AUCOMISS
+	} else {
+		checkIfMinimumSignedInt.As = x86.AUCOMISD
+	}
+	checkIfMinimumSignedInt.From.Type = obj.TYPE_MEM
+	checkIfMinimumSignedInt.From.Offset = int64(zero64BitAddress)
+	checkIfMinimumSignedInt.To.Type = obj.TYPE_REG
+	checkIfMinimumSignedInt.To.Reg = source.register
+	c.addInstruction(checkIfMinimumSignedInt)
+
+	jmpIfMinimumSignedInt := c.newProg()
+	jmpIfMinimumSignedInt.As = x86.AJCS // jump if the value is minus (= the minimum signed 32-bit int).
+	jmpIfMinimumSignedInt.To.Type = obj.TYPE_BRANCH
+	c.addInstruction(jmpIfMinimumSignedInt)
+
+	// Start emitting the error handling. These invalid flaot conversions are unrecoverable,
+	// so we must exit from the function with the dedicated status code.
+	// These existing code are jumped from cases where the original float value
+	// is either +-Inf, NaN or not in representative ranges of 32bit signed integer
+	c.addSetJmpOrigins(jmpIfExceedsLowerBound, jmpIfNaN)
+	c.setJITStatus(jitCallStatusCodeInvalidFloatToIntConversion)
+	c.returnFunction()
+
+	// We jump to the next instructions for valid cases.
+	c.addSetJmpOrigins(okJmp, jmpIfMinimumSignedInt)
+
+	// We consumed the source's register and placed the conversion result
+	// in the result register.
+	c.locationStack.markRegisterUnused(source.register)
+	loc := c.locationStack.pushValueOnRegister(result)
+	loc.setRegisterType(generalPurposeRegisterTypeInt)
+	return nil
+}
+
+// emitUnsignedI32TruncFromFloat is the implementation of compileFConvertFromI for cases
+// where the destination type is 64-bit signed integer.
+func (c *amd64Compiler) emitSignedI64TruncFromFloat(isFloat32Bit bool) error {
+	source := c.locationStack.pop()
+	if err := c.ensureOnGeneralPurposeRegister(source); err != nil {
+		return err
+	}
+
+	result, err := c.allocateRegister(generalPurposeRegisterTypeInt)
+	if err != nil {
+		return err
+	}
+
+	// First we unconditionally convert source to integer via CVTTSS2SI (CVTTSD2SI for 64bit float).
+	convert := c.newProg()
+	if isFloat32Bit {
+		convert.As = x86.ACVTTSS2SQ
+	} else {
+		convert.As = x86.ACVTTSD2SQ
+	}
+	convert.From.Type = obj.TYPE_REG
+	convert.From.Reg = source.register
+	convert.To.Type = obj.TYPE_REG
+	convert.To.Reg = result
+	c.addInstruction(convert)
+
+	// According to the Intel manual ([1],[2]), if the source float value is either +-Inf or NaN, or it exceeds representative ranges
```

Comment:
this is a big comment copy/pasted with a similar instruction. how about extracting a function for this instruction with the comment as its godoc? The parameter for that function would be different here vs 32-bit